Introduce Send pages for singlesig, single input/output send

johnny9 opened the pull request:
These changes implement the first iteration of the Send coins flow for our Desktop Wallet. A QML model, SendRecipient, is defined to hold the values entered into the Send form which will then be used to construct the actual transaction. The resulting transaction will be held in another new QML model, WalletQmlModelTransaction, and that will be held by the WalletModel and used to render the confirmation page, SendReview.qml. This is designed to closely mirror the objects used in the current Qt SendCoinsDialog modules.

johnny9 commented:
Current:

https://github.com/user-attachments/assets/bcb77901-f40d-4a58-9912-0a2b455e59cd

Goal:

https://github.com/user-attachments/assets/e7425afb-874b-4d04-8ecf-531ffdadfc77

There are a few immediate things I will be working on adjusting within the context of this PR to match the proposed designs. Primarily I will focus on the layout of the pages to match as well as the transition animations.

MarnixCroes reviewed (commented):
24d1888eead946deccf8d38b56499931a93f4637

- It happens many times that clicking the `Review` button doesn't work.
Nothing happens, it doesn't go to SendReview page.

And then after adding or removing a zero it works. (always below balance)

- The address is cleared after send is confirmed and navigated back to Send page, the `Amount` and `Note` should be cleared too?

hebasto reviewed (commented):
To fix the build [error](https://github.com/bitcoin-core/gui-qml/actions/runs/13438564685/job/37546889927):
```
RCC: Error in 'qml/bitcoin_qml.qrc': Cannot find file 'controls/LabeledTextInput.qml'
```

I suggest:
```diff
--- a/src/Makefile.qt.include
+++ b/src/Makefile.qt.include
@@ -396,6 +396,7 @@ QML_RES_QML = \
   qml/controls/ContinueButton.qml \
   qml/controls/CoreText.qml \
   qml/controls/CoreTextField.qml \
+  qml/controls/LabeledTextInput.qml \
   qml/controls/ExternalLink.qml \
   qml/controls/FocusBorder.qml \
   qml/controls/Header.qml \
```

---

Why not keep files in buildsystem scripts sorted?

hebasto commented:
@johnny9 

Want to update the PR and fix the CI?

johnny9 commented:
Update:
 - Fixed missing component in Makefile
 - Added missing bitcoinamount.h include in sendrecipient.h

johnny9 commented:
Update from [9a863b0](https://github.com/bitcoin-core/gui-qml/commit/9a863b007a5fb93745e6d15c06552711202dbb53) to [55b946a](https://github.com/bitcoin-core/gui-qml/commit/55b946aa597f85f6775ec2fab24c103270f0d7dd)
 - Rebased with main
 - Updated changes using clang-format-diff.py
 - Added missing helper function in bitcoinamount.h

johnny9 commented:
Update from [55b946a](https://github.com/bitcoin-core/gui-qml/commit/55b946aa597f85f6775ec2fab24c103270f0d7dd) to [727aa52](https://github.com/bitcoin-core/gui-qml/commit/727aa52e0576810947e34e2e0dda6ab72c3e4570)
 - Fix lint issue with duplicate include in walletqmlmodel.h

johnny9 commented:
> 24d1888eead946deccf8d38b56499931a93f4637
> 
> - It happens many times that clicking the `Review` button doesn't work.
> Nothing happens, it doesn't go to SendReview page.
> 
> And then after adding or removing a zero it works. (always below balance)
> 
> - The address is cleared after send is confirmed and navigated back to Send page, the `Amount` and `Note` should be cleared too?
> 
Both of these are good to notice and will be addressed  

The review button not working is due mainly to invalid inputs and are out of scope of the current basic implementation. This will be addressed as validation.

The form not clearing is good to note and I make an issue to address this when I come back around to validation.

johnny9 commented on src/qml/pages/wallet/Send.qml:
```diff
@@ -0,0 +1,180 @@
+                }
+            }
+
+            ContinueButton {
```
Will do. I will be coming back to this as a Validation PR.
